ArduCopter/mode_auto.cpp

```diff
@@ -2235,24 +2235,27 @@ bool ModeAuto::verify_nav_attitude_time(const AP_Mission::Mission_Command& cmd)
 // pause - Prevent aircraft from progressing along the track
 bool ModeAuto::pause()
 {
-    // do not pause if already paused or not in the WP sub mode or already reached to the destination
-    if(wp_nav->paused() || _mode != SubMode::WP || wp_nav->reached_wp_destination()) {
+    // do not pause if not in the WP sub mode or already reached to the destination
+    if(_mode != SubMode::WP || wp_nav->reached_wp_destination()) {
         return false;
     }
 
-    wp_nav->set_pause();
+    // do not pause if already paused
+    if (!wp_nav->paused()) {
```

Comment:
Looks like there is no harm in just setting it again, its just a bool in wp nav.